### PR TITLE
updated flutter_bloc and equatable dependencies

### PR DESCRIPTION
**Pull request by bartekpacia**
As the title says. It's my first PR in this project, I just want to learn the new workflow a little bit :)

### CODE REVIEW

**Comment by bartekpacia** [comment]
@isabelcosta Could I get access to merging PRs (only in this repository)? I promise I'll be responsible

**Comment by isabelcosta** [comment]
@bartekpacia you are one of the maintainers to this repository, it means that you can approve and then merge any PR that comes here from external contributors.

As a community project, we tend to put a restriction of having at least one approval required before a PR gets merged, this allows us to properly review PRs and track the changes being done to the project. Also if we create issues and PRs this provides transparency over what is happening with the project. Even, when I write code for the projects I maintain, I do changes on my fork repository and submit a PR for code review.
You just can't merge this because you are the author of the pull request if it was I doing aa PR you would be able to merge.

Do you see the PULL_REQUEST template on the .github folder? You were supposed to follow the template and give us a brief description of this change you implemented. Also, see the checklist and make sure you tick the boxes relevant to this.

Let me know what do you think about all of this, your suggestions are totally welcome here. We can even discuss all of this in a community open session, our review/merge workflow should be more clear.

**Comment by bartekpacia** [comment]
@isabelcosta This makes sense. Thank you for your exhaustive explanation. Next time I will provide much more context.
BTW I'm the one who added `PULL_REQUEST_TEMPLATE.md` into `.github` folder :)

PS So could approve this? It's a really small change and doesn't break anything. As I said, next time I'll provide more info and explanation.

**Comment by isabelcosta** [comment]
@bartekpacia I'm glad you understood me! I know you were the putting it there :P that is why I was like 🤔 
I sure can approve this, but I haven't run this yet nor I understand how to work this through. I will request review from @Techno-Disaster .

You can still edit the pull request 🙃 👉 https://github.com/anitab-org/mentorship-client/blob/develop/.github/PULL_REQUEST_TEMPLATE/pull_request.md

Can you explain at least for me to understand why you increased certain versions and decreased the library test one?

File: pubspec.lock
Comment (isabelcosta) [inline pubspec.lock:522]: this is the decrease I am talking about

Comment (isabelcosta) [reply pubspec.lock:522]: I can approve the issue and merge it, just let me know about this so I can start learning this 🤓

Comment (techno-disaster) [reply pubspec.lock:522]: @isabelcosta  pubspec.lock is a autogenerated file, but if contributors are on different channels, we might face merge conflict fue to pubspec.lock file, I have a solution in mind I'll test it and comment back soon.
Thank You

Comment (techno-disaster) [reply pubspec.lock:522]: also @isabelcosta  i think we should keep the least approvals to merge to atleast 2 people

Comment (bartekpacia) [reply pubspec.lock:522]: That's probably the case @Techno-Disaster , I was using Flutter from `master`  and when I realized it was the bleeding-edge channel I immediately switched to `stable`. IMO we should only use `stable` channel.

Comment (techno-disaster) [reply pubspec.lock:522]: iirc I use the dev channel which is the latest, like alpha channel i think we should make ot mandatory for contributors to stick to one channel to avoid thos merge conflict cc @isabelcosta

Comment (techno-disaster) [reply pubspec.lock:522]: also @isabelcosta  the students PR thier contributions directly here during their GSoC  period or they PR their whole code after the GSoC period ends?

Comment (bartekpacia) [reply pubspec.lock:522]: yes this seems like a good idea, we should create `CONTRIBUTING.md` with such info.

Comment (techno-disaster) [reply pubspec.lock:522]: yep, will make it asap, thanks for bringing this out.
So are we sticking to the master channel or dev? @bartekpacia @isabelcosta 
master : stable
dev : bleeding edge, also these builds are pretty good to, altho they are called bleeding edge i have rarely found a bug in them

Comment (bartekpacia) [reply pubspec.lock:522]: I'd lean toward `stable`, especially that we don't have any tests currently.

Comment (techno-disaster) [reply pubspec.lock:522]: we will miss some features, but if everyone is in let's stick to stable (master) channel

**Comment by isabelcosta** [comment]
> also @isabelcosta i think we should keep the least approvals to merge to atleast 2 people

@Techno-Disaster I agree here since you 2 are super active with this repository and are both committed to maintaining this. I will enable that 2 reviews.

One thing that is important though is that we have to start thinking soon in terms of a product roadmap and start prioritizing what is done here or not, so that is why I will be maintainer from a product perspective, while you (@Techno-Disaster and @bartekpacia ) are technical maintainers (since I know almost nothing of flutter).

> also @isabelcosta the students PR thier contributions directly here during their GSoC period or they PR their whole code after the GSoC period ends?

@Techno-Disaster This is a topic I am wondering, and we can discuss it at the GSoC weekly call :) They have to contribute directly to the repo, the quesiton here is that in past gsoc programs we used a dedicated branch for gsoc, and I think this should be done for the `develop` main one. But we have still to talk about this.

> So are we sticking to the master channel or dev? @bartekpacia @isabelcosta
master : stable
dev : bleeding edge, also these builds are pretty good to, altho they are called bleeding edge i have rarely found a bug in them

By channel, you 2 mean git branch? since we have `develop` and `master` branches? I welcome your suggestions here, do you want to bring this to a Zulip topic or GitHub issue? I am not sure what you two mean by stable, do you want a `stable` branch? I am totally fine with that, we just need to agree on the workflow together

**Comment by isabelcosta** [comment]
so I will rise the number of reviews here @Techno-Disaster and will approve this and leave this up to you to merge and review properly

**Comment by techno-disaster** [comment]
> By channel, you 2 mean git branch? since we have `develop` and `master` branches? I welcome your suggestions here, do you want to bring this to a Zulip topic or GitHub issue? I am not sure what you two mean by stable, do you want a `stable` branch? I am totally fine with that, we just need to agree on the workflow together


No channels in flutter are like versions, sorry for the confusion 
we have 4 channels in flutter 
master
beta
dev 
stable 
you can find more information about this here https://github.com/flutter/flutter/wiki/Flutter-build-release-channels

also according to the documentation above it says stable and beta get the exact same amount if testing so i think we should stick to beta? cc @bartekpacia

**Comment by bartekpacia** [comment]
@Techno-Disaster let's just use `stable`. We'll get new features anyway, just a bit later and better tested.

**Comment by isabelcosta** [comment]
Okay, so that whole talk about channels, could you bring this up on a different topic on Zulip under #mentorship-system stream? So that others can possibly also see this discussion about this and we can decide on the Flutter workflow :) After deciding on this, we should create documentation and setup pipeline for this.

Regarding this PR, are these files needed or not? Are they important for the project? What is your review on this @Techno-Disaster ?

**Comment by techno-disaster** [comment]
@isabelcosta if you are talking about the pubspec.lock file, yes they are needed we can't add htme to gitignore, the remaning updates to the packages are ok but thanks to @bartekpacia we came to know about this future merge conflict we could get when the coding period starts. We can now add specific instructions to are PR template so that we don't face this issue, just need to fix the template now

**Comment by bartekpacia** [comment]
it'd be awesome if you could approve/merge this @Techno-Disaster , because some Flutter VSCode extensions are yelling at me that they want `flutter_bloc==3.2.0` and `equatable==1.1.1`

**Comment by techno-disaster** [comment]
@bartekpacia  soon, once we have the complete pipeline for accepting / merging prs.

**Comment by bartekpacia** [comment]
what do you mean by saying "pipeline"? some CI/CD?

**Comment by techno-disaster** [comment]
that and fix which channel we have to use and some other small things like templates and stuff, I'm adding this as soon as a get a green signal from @isabelcosta  i have already added the pull request template, and have written half of the ci/cd script.

**Review by techno-disaster — CHANGES_REQUESTED** [review]
@bartekpacia you can now make this PR again or maybe remove the pubspec.lock changes so that the workflow action is triggered, will accept it then

**Comment by bartekpacia** [comment]
can I just force push @Techno-Disaster ?

**Comment by techno-disaster** [comment]
> can I just force push @Techno-Disaster ?

can you update this PR or close this and make a new one without the pubspec.lock? will be helpful in testing the github action

**Comment by bartekpacia** [comment]
@Techno-Disaster Okay I'll just create a new one.
Also, why no `pubspec.lock`?

**Comment by techno-disaster** [comment]
@bartekpacia It causes build errors in the action due to merge conflicts. this is checked via the git diff --exit-code command. Pubspec.lock will be manually maintained to prevent such issues

**Comment by bartekpacia** [comment]
okay
btw I have to create a new fork anyway so it's okay
PR coming in <30 mins :)

**Comment by techno-disaster** [comment]
> okay
> btw I have to create a new fork anyway so it's okay
> PR coming in <30 mins :)

cools, i suggest you do not push the pubspec.lock file to avoid errors

**Comment by bartekpacia** [comment]
well it is updated automatically  in my case

**Comment by bartekpacia** [comment]
i do `flutter pub get` and some dependencies are auto-updated

**Comment by bartekpacia** [comment]
You can't keep dependencies updated without updating `pubspec.lock`. Also, what type of errors?

**Pull request closed without merging**
